src/Dapr.Extensions.Configuration/DaprSecretStoreConfigurationExtensions.cs

```diff
@@ -44,6 +44,33 @@ public static class DaprSecretStoreConfigurationExtensions
             return configurationBuilder;
         }
 
+        /// <summary>
+        /// Adds an <see cref="IConfigurationProvider"/> that reads configuration values from the Dapr Secret Store.
+        /// </summary>
+        /// <param name="configurationBuilder">The <see cref="IConfigurationBuilder"/> to add to.</param>
+        /// <param name="store">Dapr secret store name.</param>
+        /// <param name="metadata">The secrets metadata.</param>
+        /// <param name="client">The Dapr client</param>
+        /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
+        public static IConfigurationBuilder AddDaprSecretStore(
+            this IConfigurationBuilder configurationBuilder,
+            string store,
+            DaprClient client,
+            Dictionary<string, string> metadata = null)
```

Comment:
Use `IReadOnlyDictionary<>` here and copy the doc string for `metadata` from elsewhere